functions/sendSystemMessage.ts

```diff
@@ -18,54 +17,78 @@ type EnvVars = {
 
 export type Body = {
   taskSid?: string;
+  channelSid?: string;
   message?: string;
   from?: string;
-  newStatus?: string;
   request: { cookies: {}; headers: {} };
 };
 
+export const sendSystemMessage = async (context: Context<EnvVars>, event: Body) => {
+  const { taskSid, channelSid, message, from } = event;
+
+  console.log('------ sendSystemMessage excecution ------');
+
+  if (taskSid && channelSid) {
+    return { status: 400, message: 'taskSid and channelSid both provided, exactly one expected.' };
+  }
+
+  if (taskSid === undefined && channelSid === undefined) {
+    return {
+      status: 400,
+      message: 'none of taskSid and channelSid provided, exactly one expected.',
+    };
+  }
+
+  if (message === undefined) {
+    return { status: 400, message: 'missing message.' };
+  }
+
+  const client = context.getTwilioClient();
+
+  let channelSidToMessage = null;
+
+  if (channelSid !== undefined) {
+    channelSidToMessage = channelSid;
+  } else if (taskSid !== undefined) {
```

Comment:
If check here is redundant - if you loosened up the checks above to be simple truthy / falsey checks I don't think TS would complain either.  None of the falsey values are valid for either of these inputs so it's a better check anyway